### ISSUE TITLE
Cannot read property 'colDef' of undefined with pinned column

### ISSUE THREAD

**Issue by Havock94** [angular-ui/ng-grid#3935]
Hi,
I would like to report an issue with column pinning.
It looks like sometimes (I can't reproduce the issue whenever i want) the pinned column becomes all white (until I scroll the grid, then all the values come out) and if I try to resize this column it gives me the error
Cannot read property 'colDef' of undefined
at ui.grid.min.js:12 (I couldn't find where the error is with the unminified version)

I had to unpin and then pin the column again to make it work.
I'm sorry I can't provide you more details, but I hope this report can still help you.

**Comment by Chyzwar** [comment]
Try to add enableColumnResizing: true in you column definition.

**Comment by Havock94** [comment]
Hi, thank you for your reply.
Unfortunately I already have the column resize enabled.

**Comment by Chyzwar** [comment]
Yes but you might need to try to add that in column definition not in grid
options.


2015-07-15 15:57 GMT+01:00 Havock94 <notifications@github.com>:

**Comment by SeppyB** [comment]
Hi,
also have this problem, i tried adding it as you say to the column definitions, so it looks something like :
{name: 'some column name', enableColumnResizing: true}

still getting this exception :sob:

**Comment by leelakrish** [comment]
This can be fixed by changing the findTargetCol method. if the position is left and colIndex is "0" then we need to return the "0"th element, instead of multiplying with -1 and rtlMultiplier. Please let me know if the below code fix. 

 findTargetCol: function(col, position, rtlMultiplier){
          var renderContainer = col.getRenderContainer();

          if (position === 'left') {
            // Get the column to the left of this one
            var colIndex = renderContainer.visibleColumnCache.indexOf(col);
            if(colIndex === 0){
 return renderContainer.visibleColumnCache[0];
            }else{
            return renderContainer.visibleColumnCache[colIndex - 1 * rtlMultiplier];
            }
          } else {
            return col;
          }
        }

**Comment by abhi08638** [comment]
Same issue in v4.0.6, re-adding the code above fixed it for me
